### DISCUSSION TITLE
How to Show only Images that User Has with Path Generation

### DISCUSSION THREAD

**Discussion by abbasmashaddy72** [awcodes/filament-curator#443] [Q&A] [answered]
I mean to say I have a Superadmin and a Client, where the Superadmin can have all access, but the Client can only see those items that they have uploaded using the path generator.

But Currently he is able to see all the content.

```php
 CuratorPicker::make('medias')
->hiddenLabel()
->relationship('medias', 'id')
->helperText(__('Here you can attach all files needed for this Appointment'))
->multiple()
->pathGenerator(UserPathGenerator::class)
->columnSpanFull()
->listDisplay(),
```

**Comment by awcodes** [comment]
Could possibly use a callback in the limitToDirectory() callback. Then for the resource you'd need to implement a custom policy.

**Reply by abbasmashaddy72** [reply]
Can you provide me with an example? I would be grateful to you.

**Comment by abbasmashaddy72** [comment, marked as answer]
I tried this and worked for me, I am added up an PR where we can remove the navigation which is currently not working as excepted from config:

```
CuratorPicker::make('medias')
->hiddenLabel()
->relationship('medias', 'id')
->helperText(__('Here you can attach all files needed for this Appointment'))
->multiple()
->pathGenerator(UserPathGenerator::class)
->columnSpanFull()
->directory(auth()->user()->id)
->limitToDirectory(true)
->listDisplay(),
```
